pyccel/ast/headers.py

```diff
@@ -247,20 +252,23 @@ def build_argument(var_name, dc):
                         is_argument=True)
             return var
 
-        def process_template(arg_code, Tname, d_type, type_index):
-            tmp_arg_code = arg_code.copy()
-            resolved = False
+        def process_template(signature, Tname, d_type):
+            new_sig = tuple(d_type if 'datatype' in t and t['datatype'] == Tname\
+                    else t for t in signature)
+            return new_sig
```

Comment:
Even though this is a local function, I think it's useful to add a one-line docstring to it.